sql/src/main/java/com/vmware/ddlog/DDlogHandle.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2021 VMware, Inc.
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package com.vmware.ddlog;
+
+import com.vmware.ddlog.ir.DDlogProgram;
+import com.vmware.ddlog.ir.DDlogRelationDeclaration;
+import com.vmware.ddlog.ir.DDlogTUser;
+import com.vmware.ddlog.ir.DDlogType;
+import com.vmware.ddlog.translator.RelationName;
+import com.vmware.ddlog.translator.Translator;
+import com.vmware.ddlog.util.sql.SqlStatement;
+import com.vmware.ddlog.util.sql.ToPrestoTranslator;
+import ddlogapi.*;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+
+public class DDlogHandle {
+    DDlogProgram program;
+    DDlogAPI api;
+
+    // Unfortunately, `create index` statements have to be passed separately, because neither Calcite nor Presto
+    // supports them, so we must pass them as SQL strings.
+    public <R extends SqlStatement> DDlogHandle(final List<R> ddl, final ToPrestoTranslator<R> translator,
```

Comment:
I can foresee problems with this class, which we can fix now or in a follow-up PR.

We want to avoid recompiling the DDlog program everytime we want to run benchmarks. i.e., if we know that /tmp/program.dl isn't going to change, we don't want to go through the SQL-->DDlog _and_ Rust compilation process. This _could_ be fixed with a simple boolean passed to this constructor (`compile` flag, which, if false, skips lines 56-70). 
However, this isn't enough because you've moved some DDlogProgram-specific functions into this class, such as ddlogTableTypeName, that require `this.program` to be not-null.

Another reason is, based on some idiosyncracy in DCM's workflow, we need to make a dummy "DDlogHandle", that has fields `this.program = null; this.api = null`.  So we need an empty constructor, for example.

Reply:
We can also figure out how to remove the dummy workflow.

Reply:
This PR is big enough as it is.